Kconfig

```diff
@@ -127,39 +127,9 @@ config XT_INTERRUPT_LEVEL_5
 
 rsource "src/Kconfig"
 
-choice
-	prompt "Optimization"
-	default OPTIMIZE_FOR_PERFORMANCE
-	help
-	  Controls how compiler should optimize binary.
-	  This config should affect only compiler settings and is
-	  not meant to be used for conditional compilation of code.
-
-config OPTIMIZE_FOR_PERFORMANCE
-	bool "Optimize for performance"
-	help
-	  Apply compiler optimizations prioritizing performance.
-	  It means -O2 for GCC or equivalent for other compilers.
-
-config OPTIMIZE_FOR_SIZE
-	bool "Optimize for size"
-	help
-	  Apply compiler optimizations prioritizing binary size.
-	  It means -Os for GCC or equivalent for other compilers.
-
-config OPTIMIZE_FOR_DEBUG
-	bool "Optimize for debug"
-	help
-	  Apply compiler optimizations prioritizing debugging experience.
-	  It means -Og for GCC or equivalent for other compilers.
-
-config OPTIMIZE_FOR_NONE
-	bool "Don't optimize"
-	help
-	  Apply no compiler optimizations.
-	  It means -O0 for GCC or equivalent for other compilers.
-
-endchoice
+if !ZEPHYR_SOF_MODULE
+	rsource "Kconfig.xtos-build"
```

Comment:
I fetched and tested this and it works but I don't understand how, can you add a comment explaining why `git grep ZEPHYR_SOF_MODULES` => nothing?

BTW https://docs.zephyrproject.org/latest/guides/kconfig/tips.html#lack-of-conditional-includes

Reply:
I found a few, auto-generated ZEPHYR_SOF_ variables but not this one  at https://docs.zephyrproject.org/latest/guides/modules.html

Reply:
@marc-hb Working on a reply. I just observed this is defined in my build .config and used it (and it works). But let me dig into where it's actually defined.

Reply:
Thanks! https://www.google.com/search?q=programming+by+coincidence :-)

Reply:
@marc-hb  Zephyr's build system adds this for each module. Code in here: zephyr/scripts/kconfig/kconfiglib.py
During build, e.g. ./zephyrproject/build-icl/Kconfig/Kconfig.modules is created, containing definition for CONFIG_ZEPHYR_SOF_MODULE.

Reply:
Thanks, @kv2019i can you please add a reference to `zephyr/modules/Kconfig` in a comment? I think this is not a documented feature so it deserves at least that pointer.

Reply:
> Thanks, @kv2019i can you please add a reference to zephyr/modules/Kconfig in a comment? 

Submitted in #4327